egs/wsj/s5/steps/libs/nnet3/xconfig/basic_layers.py

```diff
@@ -343,7 +343,7 @@ def get_full_config(self):
         # unlike other layers the input layers need to be printed in
         # 'init.config' (which initializes the neural network prior to the LDA)
         ans = []
-        for config_name in [ 'init', 'ref', 'final' ]:
+        for config_name in [ 'ref', 'final' ]:
```

Comment:
it will be much simpler if you stick exactly to what I sketched out.
Please revert the changes to this class.
The main thing you have to change is the function 'write_config_files' in xconfig_to_configs.py.
This will have to add code in the loop where it does:
`   for basename, lines in config_basename_to_lines.items():`
to check how many lines starting with 'output-node' each config file has.
If any config file doesn't have at least one line starting with 'output-node' then
it won't be written; and if it's not basename 'init' then this will be an error.

most of the changes you've made can be reverted.